### DISCUSSION TITLE
Publish As Azure Postgres Flexible Server Error

### DISCUSSION THREAD

**Discussion by varndellwagglebee** [microsoft/aspire#4590] [Q&A]
I have the following in my progam.cs file of the AppHost project and It's giving me an error:
Delegate 'Action<IResourceBuilder<AzurePostgresResource>, ResourceModuleConstruct, PostgreSqlFlexibleServer>' does not take 1 arguments

```C#
 var dbServer = builder.AddPostgres( "postgres" )
        .PublishAsAzurePostgresFlexibleServer(r =>
        {
            r.WithSkuName( "Standard_B1ms" );
            r.WithAutoGrowEnabled( true );
            r.WithBackupRetentionDays( 7 );
            r.WithGeoRedundantBackupEnabled( true );

        });
```

**Comment by davidfowl** [comment]
I bet copilot generated that code. That method doesn't exist. You can see it on the docs here https://learn.microsoft.com/en-us/dotnet/api/aspire.hosting.azurepostgresextensions.publishasazurepostgresflexibleserver?view=dotnet-aspire-8.0.1

There are 2 overloads and the second one takes 3 arguments.

**Comment by mitchdenny** [comment]
Moving this over to discussions since this is more of a Q&A.
